src/test/java/com/marklogic/spark/reader/file/ReadArchiveFileTest.java

```diff
@@ -0,0 +1,41 @@
+package com.marklogic.spark.reader.file;
+
+import com.marklogic.spark.AbstractIntegrationTest;
+import com.marklogic.spark.Options;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.junit.jupiter.api.Test;
+
+import java.io.UnsupportedEncodingException;
+import java.util.HashSet;
+import java.util.List;
+
+public class ReadArchiveFileTest extends AbstractIntegrationTest {
+
+    @Test
+    void readArchiveFile() throws UnsupportedEncodingException {
+        Dataset<Row> reader = newSparkSession().read()
+            .format(CONNECTOR_IDENTIFIER)
+            .option(Options.READ_FILES_TYPE, "archive")
+            .load("src/test/resources/mlcp-archive-files/sample-XML.zip");
+        verifyFileRows(reader.collectAsList());
+    }
+
+    private void verifyFileRows(List<Row> rows) throws UnsupportedEncodingException {
+        HashSet<String> hashSet = new HashSet<>();
+        hashSet.add("/alexander");
+        hashSet.add("/betsy");
+        hashSet.add("/george");
+        hashSet.add("/martha");
+        hashSet.add("/mary");
+
+        for(int i =0; i<5; i++){
+            String temp = rows.get(i).getString(0).substring(rows.get(i).getString(0).lastIndexOf("/"));
+            System.out.println("****************");
+            System.out.println(new String((byte[]) rows.get(i).get(3), "UTF-8"));
+            assert(hashSet.contains(temp));
```

Comment:
This is working and I can also see the doc content along with the corresponding metadata content. The only problem is - the content comes after the metadata content. For example, the first row  has the content for alexander.metadata followed by content in document alexander (from sample-XML.zip)